src/org/rascalmpl/interpreter/result/AbstractFunction.java

```diff
@@ -394,13 +393,13 @@ protected static Type unrenameType(Map<Type, Type> renamings, Type resultType) {
 	    return resultType;
 	}
 	  
-    public static Type renameType(Type actualTypes, Map<Type, Type> renamings) {
+    public static Type renameType(Type actualTypes, Map<Type, Type> renamings, String uniquePrefix) {
```

Comment:
if the unique prefix is always an integer, we can also make this argument an integer, and only do the `Integer.toString` inside this function?

Reply:
I changed it to full source location. the int is not unique enough if accidentally two functions come from different files, with the same offset.

Reply:
even better 👍🏼